hash_practice/exercises.py

```diff
@@ -1,20 +1,98 @@
 
+def create_freq_map(word):
+    freq_map = {}
+    for char in word:
+        if char in freq_map:
+            freq_map[char] += 1
+        else:
+            freq_map[char] = 1
+    # convert to frozenset because it's hashable whereas dicts and sets are not
+    return freq_map
+
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n^2)

```

Comment:
When doing time complexity calculations, you can't really reuse the same variables to represent different factors. For example, here, you're using `n` for both the number of strings in the list, and the number of characters in the strings, which are independent factors. So this would more properly be O(n*m) where `n` is the number of strings in the list, and `m` is the number of characters in the strings.

However, we can also make an simplifying assumption since we know our input is a list of English words. Words in English don't get too long (5 letters per word on average), so the effect of the number of characters in the word is pretty bounded, especially compared to the number of words in the list (which could easily be hundreds or thousands of words). As such, we can just say the time complexity is O(n).